compiler/modules/nback/module.bal

```diff
@@ -107,8 +138,10 @@ function createFunctionDI(ModuleDI mod, bir:File[] files, bir:FunctionDefn birFu
     });
 }
 
-function buildFunctionBody(llvm:Builder builder, Scaffold scaffold, bir:FunctionCode code) returns BuildError? {
-    foreach var b in code.blocks {
-        check buildBasicBlock(builder, scaffold, b);
+function buildFunctionBody(llvm:Builder builder, Scaffold scaffold, bir:BasicBlock[] blocks, bir:Label[] orderedLabels) returns BuildError? {
```

Comment:
I suggest you reverse the list before passing it in, so the labels give the order to build in. Then call the param `buildOrder`.